cvat-ui/src/components/analytics-page/task-quality/task-quality-component.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2023-2024 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { Row } from 'antd/lib/grid';
+import Text from 'antd/lib/typography/Text';
+import notification from 'antd/lib/notification';
+import CVATLoadingSpinner from 'components/common/loading-spinner';
+import JobItem from 'components/job-item/job-item';
+import {
+    Job, JobType, QualityReport, QualitySettings, Task, getCore,
+} from 'cvat-core-wrapper';
+import React, { useEffect } from 'react';
+import { useIsMounted, useStateIfMounted } from 'utils/hooks';
+import EmptyGtJob from './empty-job';
+import GtConflicts from './gt-conflicts';
+import Issues from './issues';
+import JobList from './job-list';
+import MeanQuality from './mean-quality';
+import QualitySettingsModal from '../shared/quality-settings-modal';
+
+const core = getCore();
+
+interface Props {
+    task: Task,
+    onJobUpdate: (job: Job) => void
+}
+
+function TaskQualityComponent(props: Props): JSX.Element {
+    const { task, onJobUpdate } = props;
+    const isMounted = useIsMounted();
+    const [fetching, setFetching] = useStateIfMounted<boolean>(true);
+    const [taskReport, setTaskReport] = useStateIfMounted<QualityReport | null>(null);
+    const [jobsReports, setJobsReports] = useStateIfMounted<QualityReport[]>([]);
+    const [qualitySettings, setQualitySettings] = useStateIfMounted<QualitySettings | null>(null);
+    const [qualitySettingsVisible, setQualitySettingsVisible] = useStateIfMounted<boolean>(false);
+    const [qualitySettingsFetching, setQualitySettingsFetching] = useStateIfMounted<boolean>(true);
```

Comment:
I added this logic before, now I have some thoughts about it:

1. We add excessive states to the component (each `useStateIfMounted` adds state and `isMounted` state). It will be alright for one component, but in perspective it will affect performance if we continue to use that this way.
2. Perhaps here we may just join all the states into one `useStateIfMounted`
3. Since there are a lot of `useState` expressions, it also may be a sign to use `useReducer` instead.